src/GafferCycles/IECoreCyclesPreview/IECoreCycles.cpp

```diff
@@ -114,9 +104,4 @@ const std::string &versionString()
 	return cyclesVersion;
 }
 
-const std::vector<ccl::DeviceInfo> &devices()
-{
-	return cyclesDevices;
```

Comment:
Long ago from what I remember the `available_devices()` call wasn't reentrant and gave different results after the first call, giving me many headaches until I realised that the Blender code only calls it once and stores the results, but maybe they've fixed that now.

Reply:
Thanks for the info on the history here Alex. In my own testing `available_devices()` seems to always return exactly the same results each time, and a glance at the Cycles source suggests that is expected (it uses a lock and flags to avoid multiple initialisations). Unless we see problems in practice, my preference is for the more minimal approach.